From 9467cbeb84defd684e15865761df401eb3323336 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Fri, 12 Apr 2024 12:11:03 +0200
Subject: [PATCH 6/7] drivers: staging: bes2600: Fix build with 6.9 kernel

Needed after https://github.com/torvalds/linux/commit/0a44dfc070749514b804ccac0b1fd38718f7daa1

DANCTNIX: The original merge request at the time is incomplete, this
fixes the error in ap.c

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 drivers/staging/bes2600/ap.c       | 8 ++++----
 drivers/staging/bes2600/main.c     | 4 ++++
 drivers/staging/bes2600/txrx_opt.c | 4 ++--
 3 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/bes2600/ap.c b/drivers/staging/bes2600/ap.c
index 61c0cb6b0f45..e0cebf9a765a 100644
--- a/drivers/staging/bes2600/ap.c
+++ b/drivers/staging/bes2600/ap.c
@@ -658,7 +658,7 @@ void bes2600_bss_info_changed(struct ieee80211_hw *dev,
 					bes2600_rate_mask_to_wsm(hw_priv,
 					sta->deflink.supp_rates[hw_priv->channel->band]));
 				rcu_read_unlock();
-				ch_type = cfg80211_get_chandef_type(&info->chandef);
+				ch_type = cfg80211_get_chandef_type(&info->chanreq.oper);
 				bes_devel("[STA] ch %d, type: %d, HT oper mode: %d\n",
 					  hw_priv->channel->hw_value,
 					  ch_type, info->ht_operation_mode);
@@ -969,7 +969,7 @@ void bes2600_bss_info_changed(struct ieee80211_hw *dev,
 	}
 
 	if (changed & BSS_CHANGED_BANDWIDTH) {
-		enum nl80211_channel_type ch_type = cfg80211_get_chandef_type(&info->chandef);
+		enum nl80211_channel_type ch_type = cfg80211_get_chandef_type(&info->chanreq.oper);
 
 		if (cfg->assoc &&
 		    hw_priv->ht_info.channel_type != ch_type &&
@@ -1548,7 +1548,7 @@ static int bes2600_start_ap(struct bes2600_vif *priv)
 	};
 
 	struct wsm_switch_channel channel = {
-		.channelMode = (cfg80211_get_chandef_type(&conf->chandef)) << 4,
+		.channelMode = (cfg80211_get_chandef_type(&conf->chanreq.oper)) << 4,
 		.channelSwitchCount = 0,
 		.newChannelNumber = hw_priv->channel->hw_value,
 	};
@@ -1560,7 +1560,7 @@ static int bes2600_start_ap(struct bes2600_vif *priv)
 
 	if (priv->vif->p2p) {
 		start_dhcpd();
-		hw_priv->ht_info.channel_type = cfg80211_get_chandef_type(&conf->chandef);
+		hw_priv->ht_info.channel_type = cfg80211_get_chandef_type(&conf->chanreq.oper);
 	}
 
 	if (priv->if_id)
diff --git a/drivers/staging/bes2600/main.c b/drivers/staging/bes2600/main.c
index ea6ba0ab170a..82b2e5efd33f 100644
--- a/drivers/staging/bes2600/main.c
+++ b/drivers/staging/bes2600/main.c
@@ -216,6 +216,10 @@ static const struct ieee80211_iface_combination bes2600_if_comb[] = {
 
 
 static const struct ieee80211_ops bes2600_ops = {
+	.add_chanctx = ieee80211_emulate_add_chanctx,
+	.remove_chanctx = ieee80211_emulate_remove_chanctx,
+	.change_chanctx = ieee80211_emulate_change_chanctx,
+	.switch_vif_chanctx = ieee80211_emulate_switch_vif_chanctx,
 	.start			= bes2600_start,
 	.stop			= bes2600_stop,
 	.add_interface		= bes2600_add_interface,
diff --git a/drivers/staging/bes2600/txrx_opt.c b/drivers/staging/bes2600/txrx_opt.c
index 2a26f737f197..6bace9d8fbf0 100644
--- a/drivers/staging/bes2600/txrx_opt.c
+++ b/drivers/staging/bes2600/txrx_opt.c
@@ -481,9 +481,9 @@ static int bes2600_set_txrx_opt_default_param(struct bes2600_common * hw_priv)
 	if (priv->join_status == BES2600_JOIN_STATUS_STA) {
 		sta = ieee80211_find_sta(priv->vif, priv->vif->bss_conf.bssid);
 		if (sta->deflink.ht_cap.ht_supported &&
-		    ((priv->vif->bss_conf.chandef.width == NL80211_CHAN_WIDTH_20 &&
+		    ((priv->vif->bss_conf.chanreq.oper.width == NL80211_CHAN_WIDTH_20 &&
 			 sta->deflink.ht_cap.cap & IEEE80211_HT_CAP_SGI_20) ||
-			(priv->vif->bss_conf.chandef.width == NL80211_CHAN_WIDTH_40 &&
+			(priv->vif->bss_conf.chanreq.oper.width == NL80211_CHAN_WIDTH_40 &&
 			 sta->deflink.ht_cap.cap & IEEE80211_HT_CAP_SGI_40))) {
 			bes_devel( "open short gi tx\n");
 			bes2600_enable_tx_shortgi(hw_priv, priv, 1);
-- 
2.45.2

